views/forms/self_referral_form.php

```diff
@@ -41,68 +41,71 @@
     $self_office_firstCall = !empty($_POST['self_office_firstCall']) ? trim($_POST['self_office_firstCall']) : NULL;
     $self_office_returnedCall = !empty($_POST['self_office_returnedCall']) ? trim($_POST['self_office_returnedCall']) : NULL;
     $self_tentative_start = !empty($_POST['self_tentative_start']) ? trim($_POST['self_tentative_start']) : NULL;
     $self_letter_mailed = !empty($_POST['self_letter_mailed']) ? trim($_POST['self_letter_mailed']) : NULL;
-    $self_assigned_to = !empty($_POST['self_assigned_to']) ? trim($_POST['self_assigned_to']) : NULL;
-    $notes = !empty($_POST['notes']) ? trim($_POST['notes']) : NULL;
+    $self_assigned_to = !empty($_POST['self_assigned_to']) ? trim($_POST['self_assigned_to']) : "";
+    $notes = !empty($_POST['notes']) ? trim($_POST['notes']) : "";
     $eID = $_SESSION['employeeid'];
 
-    // Testing variables.
-//    $arr = get_defined_vars();
-//    print_r($arr);
-
     // TODO: Stored Procedure
-//    $result = $db->query("SELECT addSelfReferral(
-//                                    fName := $1::TEXT,
-//                                    lName := $2::TEXT,
-//                                    mInit := $3::VARCHAR,
-//                                    dob := $4::DATE,
-//                                    houseNum := $5::INT,
-//                                    streetAddress := $6::TEXT,
-//                                    apartmentInfo := $7::TEXT,
-//                                    zip := $8::INT,
-//                                    cityName := $9::TEXT,
-//                                    stateName := $10::STATES,
-//                                    refSource := $11::TEXT,
-//                                    hasInvolvement := $12::BOOLEAN,
-//                                    hasAttended := $13::BOOLEAN,
-//                                    reasonAttending := $14::TEXT,
-//                                    firstCall := $15::DATE,
-//                                    returnCallDate := $16::DATE,
-//                                    startDate := $17::DATE,
-//                                    classAssigned := $18::TEXT,
-//                                    letterMailedDate := $19::DATE,
-//                                    extraNotes := $20::TEXT,
-//                                    eID := $21::INT
-//                                    );", ['$self_pers_firstname', '$self_pers_lastname', '$self_pers_middlein', $self_pers_dob,
-//                                         $self_pers_street_num, '$self_pers_street_name', '$self_apt_info', $self_pers_zip, '$self_pers_city',
-//                                         $self_pers_state,'$self_ref_source', '$self_involvement', '$self_attended','$reason', $self_office_firstCall,
-//                                         $self_office_returnedCall, $self_tentative_start, '$self_assigned_to', $self_letter_mailed, '$notes', $eID]);
+    $result = $db->query("SELECT addSelfReferral(
```

Comment:
Please look at the database as this function is now out of date. This is missing the following fields:

- referralParticipantID
- referralRace
- referralSex

Also the following fields have a different name:

| Old | New |
| --- | ---- |
| fName | referralFirstName |
| lName | referralLastName |
| mInit | referralMiddleInit |
| dob | referralDOB |

Reply:
The self referral form does not have race/sex fields on it, I'm not sure what we would enter in as values for those

Reply:
Feel free to add to this issue (https://github.com/Capping-CPCA/Database/issues/77) on the database code about the self referral insert.